test/src/materials/MaterialDerivativeTestMaterial.C

```diff
@@ -0,0 +1,43 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+#include "MaterialDerivativeTestMaterial.h"
+
+template<>
+InputParameters validParams<MaterialDerivativeTestMaterial>()
+{
+  InputParameters params = validParams<Material>();
+  params.addRequiredCoupledVar("var1", "First nonlinear variable");
+  params.addRequiredCoupledVar("var2", "Second nonlinear variable");
+  return params;
+}
+
+
+MaterialDerivativeTestMaterial::MaterialDerivativeTestMaterial(const InputParameters & parameters) :
+    DerivativeMaterialInterface<Material>(parameters),
+    _p(declareProperty<Real>("material_derivative_test_property")),
+    _dpdu(declarePropertyDerivative<Real>("material_derivative_test_property", "u")),
```

Comment:
i think it is better to change `u` to `getVar("var1")->name()` and `v` similarly for consistency.